nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CountText.java

```diff
@@ -0,0 +1,318 @@
+import org.apache.nifi.annotation.behavior.SideEffectFree;
+import org.apache.nifi.annotation.behavior.SupportsBatching;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.behavior.WritesAttributes;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.SeeAlso;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.util.StringUtils;
+
+@EventDriven
+@SideEffectFree
+@SupportsBatching
+@Tags({"count", "text", "line", "word", "character"})
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@CapabilityDescription("Counts various metrics on incoming text. The requested results will be recorded as attributes. "
+        + "The resulting flowfile will not have its content modified.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "text.line.count", description = "The number of lines of text present in the FlowFile content"),
+        @WritesAttribute(attribute = "text.line.nonempty.count", description = "The number of lines of text (with at least one non-whitespace character) present in the original FlowFile"),
+        @WritesAttribute(attribute = "text.word.count", description = "The number of words present in the original FlowFile"),
+        @WritesAttribute(attribute = "text.character.count", description = "The number of characters (given the specified character encoding) present in the original FlowFile"),
+})
+@SeeAlso(SplitText.class)
+public class CountText extends AbstractProcessor {
+    private static final List<Charset> STANDARD_CHARSETS = Arrays.asList(
+            StandardCharsets.UTF_8,
+            StandardCharsets.US_ASCII,
+            StandardCharsets.ISO_8859_1,
+            StandardCharsets.UTF_16,
+            StandardCharsets.UTF_16LE,
+            StandardCharsets.UTF_16BE);
+
+    private static final String SYMBOL_REGEX = "[\\s-\\._]";
+    private static final String WHITESPACE_ONLY_REGEX = "\\s";
+
+    // Attribute keys
+    public static final String TEXT_LINE_COUNT = "text.line.count";
+    public static final String TEXT_LINE_NONEMPTY_COUNT = "text.line.nonempty.count";
+    public static final String TEXT_WORD_COUNT = "text.word.count";
+    public static final String TEXT_CHARACTER_COUNT = "text.character.count";
+
+
+    public static final PropertyDescriptor TEXT_LINE_COUNT_PD = new PropertyDescriptor.Builder()
+            .name("text-line-count")
+            .displayName("Text Line Count")
```

Comment:
Given that these are boolean properties, I wonder if it makes more sense to word as "Count Lines", "Count Non-Empty Lines", "Count Words"?
Is a minor point, and if you don't agree it's okay to keep as-is but I found the wording less intuitive personally. Up to you.

Reply:
I can change these to be clearer as you suggested.